README.md
I modified http://ghost.tweakblogs.net/blog/3073/subversion-repository-monitor.html to show
commit messages and changed how often it runs. Furthermore, now multiple repositories are supported.

run following command to build a debian package
> debuild -b -us -uc




debian/changelog
svnnotify (0.4-1) UNRELEASED; urgency=low

  * added desktop icon
  * split into two files

 -- mog33 <ctrl.alt.del@gmx.net>  Mon, 05 Aug 2013 21:59:51 +0200

svnnotify (0.3-1) unstable; urgency=low

  * Minor bugfixes
  * correct path as working directory
  * using debhelper for creating debian package

 -- mog33 <ctrl.alt.del@gmx.net>  Tue, 30 Jul 2013 08:58:31 +0200

svnnotify (0.2-1) unstable; urgency=low

  * Initial release 

 -- mog33 <ctrl.alt.del@gmx.net>  Mon, 29 Jul 2013 18:52:36 +0200


go.py
#!/usr/bin/python
#
# script to notify the user for changes in a subversion repository.
#
# Depends on a configuration file with the following entries:
#
#   [server]
#   server=SVN_REPO_TO_MONITOR
#   user=YOUR_SVN_USERNAME
#   pass=YOUR_SVN_PASSWORD
#

try:
    import pysvn, pynotify
except:
    print "Error while loading external depencencies."
    print "Make sure 'pysvn' and 'pynotify' is installed."
    exit()

import datetime, time, os, ConfigParser as cfg

class SvnNotifier():

    def __init__(self, name, svn_root, svn_username, svn_password):
        self.name = name
        self.svn_root = svn_root
        self.svn_username = svn_username
        self.svn_password = svn_password
        self.last_revision = pysvn.Revision(pysvn.opt_revision_kind.number, 0)
        self.client = pysvn.Client()
        self.client.callback_get_login = self.credentials

    def notify(self, author, date, revision, message, paths):
        """Display the changed paths using libnotify"""
        title_string = 'New commit #%s to repository %s' % (revision, self.name)
        message_string = "[%s] %s: %s\n Paths: %s" % \
                        (date.strftime("%d.%m %H:%M"), author, message, paths)
        pynotify.Notification(title_string, message_string, "view-refresh").show()

    def credentials(self, realm, username, may_save):
        """Return the default login credentials"""
        return True, self.svn_username, self.svn_password, False

    def discover_changes(self):
        """
        Find out the changes occured since the last time this method is ran
        """
        log = self.client.log(
            self.svn_root,
            discover_changed_paths=True,
            revision_end=self.last_revision
            )
        log = log[:-1]   # Ignore last revision
        if len(log) > 0:
            self.last_revision = log[0].revision
            if len(log) > 3:
                pynotify.Notification("Older Commits for repository %s available" % self.name, None, "view-refresh").show()
                log = log[0:3]
            log.reverse()
            for entry in log:
                author = entry.get('author')
                rev = entry.revision.number
                message = entry.message
                date = datetime.datetime.fromtimestamp(entry.date)
                paths = []
                for change in entry.changed_paths:
                    if change.path not in paths:
                        paths.append(change.path)
                self.notify(author, date, rev, message, paths)


def main():
    path = os.path.dirname(__file__)
    config_file = os.path.join(path, 'svnnotify.cfg')
    repos = []
    try:
        parser = cfg.ConfigParser()
        parser.read(config_file)
        for section in parser.sections():
            server = parser.get(section, 'server')
            user = parser.get(section, 'user')
            passw = parser.get(section, 'pass')
            repos.append(SvnNotifier(section, server, user, passw))
            print 'Monitoring SVN repository %s: %s' % (section, server)
    except BaseException as e:
        print "Error while parsing file '%s':" % config_file
        print e
        exit()
    pynotify.init("SVN Monitor")
    print '- Press %s to quit -' % '^C'
    while True:
        for repo in repos:
            repo.discover_changes()
        time.sleep(60 * 10)  # Every 10 minutes

if __name__ == '__main__':
    main()


SvnRepoMonitor.py
import pynotify
import datetime, logging
from pysvn import Client, Revision, opt_revision_kind
from ConfigParser import ConfigParser


class SvnRepoMonitor():

    def __init__(self, name, svn_root, svn_username, svn_password, config_file):
        self.name = name
        self.svn_root = svn_root
        self.svn_username = svn_username
        self.svn_password = svn_password
        self.client = Client()
        self.client.callback_get_login = self.credentials
        self.parser = ConfigParser()
        self.config_file = config_file

    def notify(self, author, date, revision, message, paths):
        """Display the changed paths using libnotify"""
        title_string = 'New commit #%s in repository %s' % (revision, self.name)
        message_string = "<p>[%s] %s</p> <p><i>%s</i>" % \
                        (date.strftime("%d.%m %H:%M"), author, message)
        message_string += "<ul>"
        for p in paths[:5]:
            if len(p)>50:
                p = "...%s" % p[-50:]
            message_string += "<li>%s</li>" % p
        if len(paths)>6:
            message_string += "<li>...</li>"
        message_string += "</ul></p>"
        logging.debug("Open pynotify.Notification: %s | %s" % (title_string, message_string))
        pynotify.Notification(title_string, message_string, "view-refresh").show()

    def credentials(self, realm, username, may_save):
        """Return the default login credentials"""
        return True, self.svn_username, self.svn_password, False

    def discover_changes(self):
        """
        Find out the changes occured since the last time this method is ran
        """
        logging.debug("Discover Changes in %s" % self.name)
        self.parser.read(self.config_file)
        if self.parser.has_option(self.name, "last_revision"):
            last_revision = self.parser.get(self.name, "last_revision")
        else:
            last_revision = 0
        log = self.client.log(
            self.svn_root,
            discover_changed_paths=True,
            revision_end=Revision(opt_revision_kind.number, last_revision)
            )
        log = log[:-1]   # Ignore last revision
        if len(log) > 0:
            logging.info("%s new commits in repository %s" % (len(log), self.name))
            # update last revision in config file
            last_revision = log[0].revision.number
            self.parser.set(self.name, "last_revision", last_revision)
            self.parser.write(open(self.config_file, 'w'))
            log.reverse()
            if len(log)>5: # Show only most recent commits
                pynotify.Notification("Even more commits in repository %s" % self.name, "There are %s more new commits in the repository" % (len(log)-5), "view-refresh").show()
                log = log [-5:]
            for entry in log:
                author = entry.get('author')
                rev = entry.revision.number
                message = entry.message
                date = datetime.datetime.fromtimestamp(entry.date)
                paths = []
                for change in entry.changed_paths:
                    paths.append(change.path)
                self.notify(author, date, rev, message, paths)
        else:
            logging.debug("No new commits in repository %s" % self.name)
 

svnnotify.py
#!/usr/bin/python
#
# Script to notify the user for changes in a subversion repository.
#
# Depends on a configuration file with one or more of the following entries:
#
#   [ServerName]
#   server=SVN_REPO_TO_MONITOR
#   user=YOUR_SVN_USERNAME
#   pass=YOUR_SVN_PASSWORD
#

import pynotify
import time, os, logging
from appdirs import user_data_dir
from ConfigParser import ConfigParser
from SvnRepoMonitor import SvnRepoMonitor

class SvnNotifier():

    def __init__(self, config_file, interval):
        logging.basicConfig(
            level = logging.INFO,
            format = "%(asctime)s %(levelname)s: %(message)s",
            datefmt = "%d.%m.%Y %H:%M:%S",
            #filename= os.path.join(os.getcwd(), 'svnnotify.log')
            )
        self.interval = interval
        if config_file:
            self.config_file = config_file
        else:
            self.config_file = self.get_config_file()
        self.repos = self.create_repos_from_svn_config()
        pynotify.init("SVN Monitor")
        logging.info("SvnNotifier created with configuration file %s monitoring %i repositories." % (self.config_file, len(self.repos)))

    def run(self):
        while True:
            for repo in self.repos:
                try:
                    repo.discover_changes()
                except ClientError as e:
                    logging.error("Repo %s not accessible\n%s" % (repo.name,e))
            time.sleep(self.interval)

    def create_repos_from_svn_config(self):
        logging.info("Reading configuration file %s" % self.config_file)
        parser = ConfigParser()
        parser.read(self.config_file)
        repos = []
        for section in parser.sections():
            try:
                server = parser.get(section, 'server')
            except BaseException as e:
                logging.critical("Error while parsing config file %s\n%s" % (self.config_file,e))
                exit()
            if parser.has_option(section, 'user'):
                user = parser.get(section, 'user')
            else:
                user = None
            if parser.has_option(section, 'pass'):
                passw = parser.get(section, 'pass')
            else:
                passw = None
            repos.append(SvnRepoMonitor(section, server, user, passw, self.config_file))
            logging.info('Monitoring SVN repository %s (%s)' % (section, server))
        if repos:
            return repos
        else:
            logging.error("No sections in configuration file found. Aborting")
            exit()

    def get_config_file(self):
        cfg = os.path.join(os.getcwd(),"svnnotify.cfg")
        if os.path.isfile(cfg):
            return cfg
        cfg = os.path.join(user_data_dir("svnnotify"),"svnnotify.cfg")
        if os.path.isfile(cfg):
            return cfg
        logging.error("No configuration file found. Aborting")
        exit()


if __name__ == '__main__':
    from argparse import ArgumentParser, FileType
    parser = ArgumentParser(description="Notifies about changes in SVN repositories")
    parser.add_argument("-c","--config", help="path to configuration file (default svnnotify.cfg in working directory or in %s)" % user_data_dir("svnnotify"))
    parser.add_argument("-v","--version", help="shows version of svnnotify", action='version', version='%(prog)s 0.4')
    parser.add_argument("-i","--interval", help="update interval in seconds (default 5 minutes)", type=int, default=60*5)
    args = parser.parse_args()
    svn_notifier = SvnNotifier(args.config, args.interval)
    svn_notifier.run()
